samza-kafka/src/main/scala/org/apache/kafka/clients/consumer/KafkaConsumerConfig.java

```diff
@@ -0,0 +1,180 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.apache.kafka.clients.consumer;
+
+import java.util.Map;
+import java.util.Properties;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.samza.SamzaException;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.ConfigException;
+import org.apache.samza.config.JobConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Option;
+
+
+/**
+ * The configuration class for KafkaConsumer
+ */
+public class KafkaConsumerConfig extends ConsumerConfig {
+
+  public static final Logger LOG = LoggerFactory.getLogger(KafkaConsumerConfig.class);
+
+  private static final String PRODUCER_CLIENT_ID_PREFIX = "kafka-producer";
+  private static final String CONSUMER_CLIENT_ID_PREFIX = "kafka-consumer";
+  private static final String ADMIN_CLIENT_ID_PREFIX = "samza-admin";
+  private static final String SAMZA_OFFSET_LARGEST = "largest";
+  private static final String SAMZA_OFFSET_SMALLEST = "smallest";
+  private static final String KAFKA_OFFSET_LATEST = "latest";
+  private static final String KAFKA_OFFSET_EARLIEST = "earliest";
+  private static final String KAFKA_OFFSET_NONE = "none";
+
+  /*
+   * By default, KafkaConsumer will fetch ALL available messages for all the partitions.
+   * This may cause memory issues. That's why we will limit the number of messages per partition we get on EACH poll().
+   */
+  private static final String KAFKA_CONSUMER_MAX_POLL_RECORDS_DEFAULT = "100";
+
+
+  public KafkaConsumerConfig(Properties props) {
+    super(props);
+  }
+
+  public static KafkaConsumerConfig getKafkaSystemConsumerConfig(Config config, String systemName, String clientId,
+      Map<String, String> injectProps) {
+
+    Config subConf = config.subset(String.format("systems.%s.consumer.", systemName), true);
+
+    String groupId = getConsumerGroupId(config);
+
+    Properties consumerProps = new Properties();
+    consumerProps.putAll(subConf);
+
+    consumerProps.setProperty(ConsumerConfig.GROUP_ID_CONFIG, groupId);
+    consumerProps.setProperty(ConsumerConfig.CLIENT_ID_CONFIG, clientId);
+
+    //Kafka client configuration
+
+    // Disable consumer auto-commit because Samza controls commits
+    consumerProps.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+
+    // Translate samza config value to kafka config value
+    consumerProps.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG,
+        getAutoOffsetResetValue(consumerProps));
+
+    // make sure bootstrap configs are in ?? SHOULD WE FAIL IF THEY ARE NOT?
+    if (!subConf.containsKey(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG)) {
```

Comment:
Previously we were using SimpleConsumer in samza which had configurations (such as `zookeeper.connect`) which are no longer supported in the new kafkaConsumer.

New kafka consumer configs: http://kafka.apache.org/documentation/#newconsumerconfigs. 

We need to update our samza configuration table. 

1. Delete the old configuration references (systems.system-name.consumer.zookeeper.connect, etc) 
2. Stress some of the important features of new kafkaConsumer(`SSL`, `keySerializer`, `valueSerializer`, `max.poll.records`). Our implementation overrides the defaults with user-defined configurations and it will be better to document it.